modelling/src/main/java/org/axonframework/modelling/entity/child/CommandTargetResolver.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.modelling.entity.child;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.List;
+
+/**
+ * Resolves the target child entity for a given command message. This is used to determine which child entity should
+ * handle a command when one multiple child entities are present.
+ *
+ * @param <E> The type of child entity to resolve.
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+@FunctionalInterface
+public interface CommandTargetResolver<E> {
```

Comment:
This reminds me of something: https://github.com/AxonFramework/AxonFramework/blob/master/modelling/src/main/java/org/axonframework/modelling/command/CommandTargetResolver.java

Reply:
Yeah, even though the name is the same, it functions quite differently (one is for the root entity, the other for children).  The original `CommandTargetResolver` from 4, was actually replaced by the `EntityIdResolver` in an earlier PR. As such, I think this is quite a different component although the name is the same. Wdyt?

Reply:
Having it drafted in the `api-changes.md` is indeed what I was aiming for :-)
I think we can also deprecate the old `CommandTargetResolver`...I'll see in a bit, during my re-review, whether that's still outstanding.